plugins/teststeps/sshcmd/sshcmd.go

```diff
@@ -186,31 +201,49 @@ func (ts *SSHCmd) Run(cancel, pause <-chan struct{}, ch test.TestStepChannels, p
 			errCh <- innerErr
 		}()
 
-		select {
-		case err := <-errCh:
-			log.Infof("Stdout of command '%s' is '%s'", cmd, stdout.Bytes())
-			if err == nil {
-				// Execute expectations
-				expect := ts.Expect.String()
-				if expect == "" {
-					log.Warningf("no expectations specified")
-				} else {
-					re := regexp.MustCompile(expect)
-					matches := re.FindAll(stdout.Bytes(), -1)
-					if len(matches) > 0 {
-						log.Infof("match for regex \"%s\" found", expect)
+		expect := ts.Expect.String()
+		for {
+			select {
+			case err := <-errCh:
+				log.Infof("Stdout of command '%s' is '%s'", cmd, stdout.Bytes())
+				if err == nil {
+					// Execute expectations
+					if expect == "" {
+						log.Warningf("no expectations specified")
 					} else {
-						return fmt.Errorf("match for %s not found for target %v", expect, target)
+						re := regexp.MustCompile(expect)
+						matches := re.FindAll(stdout.Bytes(), -1)
+						if len(matches) > 0 {
+							log.Infof("match for regex \"%s\" found", expect)
+						} else {
+							return fmt.Errorf("match for %s not found for target %v", expect, target)
+						}
 					}
+				} else {
+					log.Warningf("Stderr of command '%s' is '%s'", cmd, stderr.Bytes())
+				}
+				return err
+			case <-cancel:
+				return session.Signal(ssh.SIGKILL)
+			case <-pause:
+				return session.Signal(ssh.SIGKILL)
+			default:
+				re := regexp.MustCompile(expect)
+				matches := re.FindAll(stdout.Bytes(), -1)
+				if len(matches) > 0 {
+					log.Infof("match for regex \"%s\" found", expect)
+					return nil
+				}
+				if time.Now().After(timeTimeout) {
+					return fmt.Errorf("timed out after %s", timeout)
+				}
+				err = session.Signal(ssh.Signal("CONT"))
```

Comment:
can you add a comment on why this is necessary?

Reply:
Ack

Reply:
this will generate quite a bit of activity on the connection, we'll be sending and receiving 4 packets every second.
do you know why the connection gets dropped? is it inactivity or what?

Reply:
Not sure.

I tried working around this with ServerAliveInterval - The BMC (OpenBMC) but actually requires "keystrokes" to not shut down the connection to the client. Thus I'm sending CONT to keep the connection. IMHO 4 CONT's a second isn't too much noise - but I do see the point.

We can work around this by skipping this - or only sending CONT every x times. Otherwise I can also leave it out and have it sitting in my own fork until we worked around this.

Reply:
I can also guard this with a flag - if that's better.

Reply:
let's do every 20th time, this way we only send every 5 seconds

Reply:
Done.

Reply:
you forgot to ++ the counter

Reply:
:D - it's 01:49am..

Reply:
Done. Forced-Push - never happend.

Reply:
:)